src/main/java/Duke/Chatbot/Ui.java

```diff
@@ -46,68 +70,128 @@ public void showStartupMessage() {
         System.out.println(MESSAGE_START);
     }
 
+    /**
+     * Message to be shown to user when message is successfully removed from the list
+     *
+     * @param taskMessage User representation of the message successfully removed
+     * @param taskLength  Length of the list after this removal
+     */
     public void showRemovedTaskMessage(String taskMessage, int taskLength) {
         System.out.println("Got it. I've removed this task:");
         showTaskEditedMessage(taskMessage, taskLength);
     }
 
+    /**
+     * Shows the User representation of the tasks that are currently in the list
+     *
+     * @param tasks User representation of the tasks that are currently in the list
+     */
     public void showTasksMessage(String tasks) {
         System.out.println("Here are the tasks in your list:");
         System.out.println(tasks);
     }
 
+    /**
+     * Error message to be shown when task the user is trying to mark
+     * has already been marked
+     */
     public void showTaskStateCompletedError() {
```

Comment:
I find it nice that you added more specific errors to be more clear to the user.